render-only-sample/rosumd/RosUmdResource.cpp

```diff
@@ -449,24 +543,151 @@ bool RosUmdResource::CanRotateFrom(const RosUmdResource* Other) const
            (m_hwHeightTiles == Other->m_hwHeightTiles);
 }
 
+
+// Converts R, RG or A buffer to 32 bpp (RGBA) buffer
+void  RosUmdResource::ConvertBufferto32Bpp(BYTE *pSrc, BYTE *pDst, UINT srcBpp, UINT swizzleMask, UINT pSrcStride, UINT pDstStride)
+{
+    for (UINT i = 0; i < m_mip0Info.TexelHeight; i++)
+    {
+        DWORD *dstSwizzled = (DWORD*)pDst;
+
+        UINT dstIndex = 0;
+        for (UINT k = 0; k < m_mip0Info.TexelWidth*srcBpp; k += srcBpp)
+        {
+            DWORD swizzledRGBA = 0;
+
+            // Gather individual color elements into one DWORD
+            for (UINT colorElement = 0; colorElement < srcBpp; colorElement++)
+            {
+                DWORD currentColorElement = (DWORD)pSrc[k + colorElement];
+
+                // Move element to the right position
+                currentColorElement = currentColorElement << (colorElement << 3);
+                swizzledRGBA = swizzledRGBA | currentColorElement;
+            }
+
+            swizzledRGBA = swizzledRGBA << swizzleMask;
+            dstSwizzled[dstIndex] = swizzledRGBA;
+            dstIndex += 1;
+        }
+
+        pSrc += pSrcStride;
+        pDst += pDstStride;
+    }
+}
+
+// Converts texture to internal (HW friendly) representation
+void RosUmdResource::ConvertInitialTextureFormatToInternal(BYTE *pSrc, BYTE *pDst, UINT rowStride)
+{
+    // HW supports only limited set of formats, so most of DXGI formats must
+    // be converted at the beginning.    
+    UINT swizzleMask = 0;
```

Comment:
Consider to leave uninitialized so that compiler may detect code paths which fail to initialize the variables properly.